src/packages/piral-cli/src/parcel/bundler.ts

```diff
@@ -157,7 +157,7 @@ export async function patchModules(rootDir: string, ignoredPackages = defaultIgn
 }
 
 const bundleUrlRef = '__bundleUrl__';
-const piletMarker = '//@pilet v:';
+const piletMarker = '/*@pilet v:';
```

Comment:
This we cannot accept. The marker is wired to the feed service and according to our spec its a single line.

Reply:
Then the only other option is to postprocess the map files after piral performs a build, isn't it?

There are also some issues with map files (built by tsc) for other packages as their namespace aren't descriptive enough and are clashing between the packages. 

These sourcemap issues makes it next to impossible to use a debugger. 

Would there be any interest in establishing such postprocessing of the sourcemaps?

Reply:
Yes definitely!

We've been discussing this internally already. The reason for the PR (shift in offset) could be solved potentially by an algorithm / implementation similar to [this lib](https://github.com/twada/offset-sourcemap-lines). If the lib matches (I could not have a detailed look, from the outside it seems fine) we could just use it.

For the other problem regarding TSC I don't know - if we can do something about it we should. Can you give an example and how it would look improved? We would love to improve source maps support also for Parcel (as a fact we already improved it in the Webpack case).

Essentially, we would the same `postProcess` method that is used for doing the changes to the pilet. We already identify other child bundles (CSS), so identifying and modifying `.js.map` should be possible, too.

What are your thoughts?

Reply:
My first thought that indeed ideally Parcel would allow injecting some custom build steps, and then it would be producing sourcemaps after those custom steps are done. As a quick fix (=hack) it could maybe accept an extra offset (defaulting to zero) as part of its input configuration.

But then, I thought this will take to long to get it done, so I attempted to produce the least disruptive (but probably not very correct) changes, that would "work for me (c)" :)

I agree that postprocessing seems the best option, especially since there are those troubles with tsc maps too. Here is the typical map produced by tsc:
![image](https://user-images.githubusercontent.com/48287147/81614870-f88a6c00-93e0-11ea-8e3f-2d941be13523.png)
After manually tweaking the path to:
![image](https://user-images.githubusercontent.com/48287147/81614923-0dff9600-93e1-11ea-9377-f12b186b16f9.png)

Chrome would display the file inside "node_modules" (while previously piral-auth/create.ts was clashing with piral-base/create.ts), ensuring each package has its own directory:
![image](https://user-images.githubusercontent.com/48287147/81615010-31c2dc00-93e1-11ea-8f13-ccc41b214116.png)

By tweaking those pathes the place where the source file is placed in the debugger could be virtually anywhere. I believe the postprocess job could be performing such task.

Reply:
My first thought that indeed ideally Parcel would allow injecting some custom build steps, and then it would be producing sourcemaps after those custom steps are done. As a quick fix (=hack) it could maybe accept an extra offset (defaulting to zero) as part of its input configuration.

But then, I thought this will take to long to get it done, so I attempted to produce the least disruptive (but probably not very correct) changes, that would "work for me (c)" :)

I agree that postprocessing seems the best option, especially since there are those troubles with tsc maps too. Here is the typical map produced by tsc:
![image](https://user-images.githubusercontent.com/48287147/81615733-61beaf00-93e2-11ea-9b6b-73cddfecdc27.png)

After manually tweaking the path to:
![image](https://user-images.githubusercontent.com/48287147/81614923-0dff9600-93e1-11ea-9377-f12b186b16f9.png)

Chrome would display the file inside "node_modules" (while previously piral-auth/create.ts was clashing with piral-base/create.ts), ensuring each package has its own directory:
![image](https://user-images.githubusercontent.com/48287147/81615010-31c2dc00-93e1-11ea-8f13-ccc41b214116.png)

By tweaking those pathes the place where the source file is placed in the debugger could be virtually anywhere. I believe the postprocess job could be performing such task.

Reply:
Yeah I see what you mean. Thanks for pointing that out!

These paths are indeed quite annoying. Maybe we could go for something like what Webpack does with its own protocol specifier `webpack://`? Not sure how exactly it works under the hood, but in the end it produces a much nicer debugging experience.

Reply:
So should we try to include the improved source map handling in this PR?

We could use the `source-map` [lib](https://www.npmjs.com/package/source-map) for all modifications. If follows the standard and has a useable API.

Would you love to try making a first draft (in this PR)? Would be much appreciated :beers!

Reply:
Haha, sure! I would like to fix those tsc files too :)
That source-map library seems like a good option.

I can't promise though that those changes will happen soon - might take several weeks for me :)

Reply:
I've pushed a change that changes the offset. Maybe the filenames can still be improved. Any contribution would be awesome!

According to https://sokra.github.io/source-map-visualization the source map is now correct.